### PR TITLE
Adopt UV in GHA

### PR DESCRIPTION
**Pull request by FBruzzesi**
# Description

Not necessary but a nice to have for such minimal change.
Moves from `pip` to [`uv`](https://astral.sh/blog/uv) in Github Actions

## Details

Installing dependencies has a speedup between 2x and 4x, and overall running time between 1.1x and 2.4x

### CODE REVIEW

File: .github/workflows/schedule-dependencies.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 name: Cron Test Dependencies
 
 on:
+  workflow_dispatch:
```
Comment (FBruzzesi) [inline .github/workflows/schedule-dependencies.yml:4]: This is highly unnecessary, but maybe come useful for testing --pre on a specific branch before merging.
~We~ I don't have another way of running it

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,24 +1,25 @@
-Before working on a large PR, please check with @koaning or @MBrouns that they agree with the direction of the PR. This discussion should take place in a Github issue before working on the PR, unless it's a minor change like spelling in the docs. 
+Before working on a large PR, please check with @koaning, @MBrouns or @FBruzzesi that they agree with the direction of the PR. This discussion should take place in a [Github issue](https://github.com/koaning/scikit-lego/issues/new/choose) before working on the PR, unless it's a minor change like spelling in the docs.
```
Comment (koaning) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
Before working on a large PR, please check with @FBruzzesi or @koaning to confirm that they agree with the direction of the PR. This discussion should take place in a [Github issue](https://github.com/koaning/scikit-lego/issues/new/choose) before working on the PR, unless it's a minor change like spelling in the docs. 
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
-- [ ] My code follows the style guidelines (flake8)
+- [ ] My code follows the style guidelines (ruff)
 - [ ] I have commented my code, particularly in hard-to-understand areas
 - [ ] I have made corresponding changes to the documentation (also to the readme.md)
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] I have added tests to check whether the new feature adheres to the sklearn convention
 - [ ] New and existing unit tests pass locally with my changes
 
-If you feel your PR is ready for a review, ping @koaning or @mbrouns. 
+If you feel your PR is ready for a review, ping @koaning, @mbrouns or @FBruzzesi.
```
Comment (koaning) [inline .github/PULL_REQUEST_TEMPLATE.md:25]:
```suggestion
If you feel your PR is ready for a review, ping @FBruzzesi or @koaning.
```

**Review by koaning — APPROVED** [review]
Looks good. Left some minor suggestions because @MBrouns is certainly less active here these days. Feel free to merge once branch looks green.

**Pull request merged**